src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3352,7 +3370,13 @@ bool DoorLockServer::HandleRemoteLockOperation(chip::app::CommandHandler * comma
 exit:
     if (!success && reason == OperationErrorEnum::kInvalidCredential)
     {
-        TrackWrongCodeEntry(endpoint);
+        HandleWrongCodeEntry(endpoint);
+    }
+
+    // Reset the wrong code retry attempts if a valid credential is presented during lock/unlock
+    if ((opType == LockOperationTypeEnum::kUnlock || opType == LockOperationTypeEnum::kLock) && success && pinCode.HasValue())
```

Comment:
```suggestion
    if (success && pinCode.HasValue())
```
The opType here is always either Lock or Unlock.

Reply:
There are 2 other codes in the enum (not valid op types though). chances are it won't be called for those but to be safe

Reply:
See the VerifyOrDie at the very top of the function.  And if _that_ ever changes, then the function needs to be read very carefully to make sure it's ok with the new optypes, top to bottom, not just this part.

Reply:
i missed that. let me check